### DISCUSSION TITLE
2023-06-9 - Databases, data warehouses, and data lakes - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by lukecookssw** [SSWConsulting/SSW.Rules.Content#5758] [Chewing the Fat]
Hey data engineers!

Let's talk about [this rule](https://www.ssw.com.au/rules/use-the-right-data-storage/)

**Comment by leehawkinsssw** [comment]
I spent 20+ years working on projects building database tools at Quest, so a lot of this was familiar ground. Quest were super early into NoSQL databases (too early for the market to be there in fact), but I didn't encounter data lakes so the rule (and video) provided some useful additional information for me.

On the CTF form, the multiple choice options are irksome, "Somewhat not confident" and "Extremely not confident" is very clumsy English.

**Comment by bradystroud** [comment]
Data lake and data warehouse are bad names, they don’t explain what they are.

https://chat.openai.com/share/6af60e05-b164-4ada-ad9b-47c73cf6e74a

**Reply by jakebayliss** [reply]
Data oasis!

**Reply by danielmackay** [reply]
Data swamp!

**Reply by Geordie88** [reply]
I like data lake but I agree that warehouse means nothing

**Reply by leehawkinsssw** [reply]
"There are many dangers skulking in and around lakes, but if we look for the warning signs and [make wise decisions](https://www.redcross.org/get-help/how-to-prepare-for-emergencies/types-of-emergencies/water-safety/lake-river-safety.html) to avoid these areas and situations, perhaps these dangers would not be a factor when we find a great lake to dip in." https://unlockoutdoors.com/lake-swimming-dangers/

**Comment by jakebayliss** [comment]
Without having used a data lake before they just seem like they would be very messy and give me the shivers. I will need to see it in action!

**Reply by brydeno** [reply]
They require discipline. Many of them have logical structures (containers), so you can keep cleaner more structured data away from the pollution of all the other data you've added.

**Comment by tkapa** [comment]
Data and interfacing with it is interesting to me, but I haven't had much opportunity to interact with stuff like this at scale (making a DB with 100+ tables for me is wild).

I think the rule is a big wall of text, and could probably be summarised a bit better.

**Comment by leahy268** [comment]
Did lots of data analysis over the years.  Mostly I did all the ETL just directly in SQL views and then into PowerBI, often bringing multiple other sources, csv's and json data in.  Unfortunately at the time it was working for a school which wouldn't have justified the extra cost associated with a datawarehouse.  Super excited to look at Fabric..

**Comment by MattParkerDev** [comment]
I’m interested in learning more about data warehouses and Lakes, I haven’t used them before.

**Comment by 2fernandez** [comment]
Usually, designers dont have great knowledge about big, small, midget data and so on... it was good to see this video and educate myself about data lakes. It gives me some perspective on how developers might be handling data, hopefully, this information will help me with a UX process at some point.

Now for the weekly meme:
![Unknown](https://github.com/SSWConsulting/SSW.Rules.Content/assets/27670973/c52f819f-a8f2-4178-abb8-cc492390e251)

**Reply by lukecookssw** [reply]
>  midget data

I believe the term is "vertically-challenged data"

**Comment by ncn-ssw** [comment]
Given that Data Lakes require "proper governance and metadata management", there is likely to be a lot of business arising from the need to rectify the mistakes of early adopters.

**Comment by UlyssesMaclaren** [comment]
I loved the video that explained it in a really accessible way :)

**Comment by AshAnil** [comment]
The rule and video are good ⭐I would like to get hands-on to work on creating databases.

**Comment by camillars** [comment]
Video + rule made it easy to understand the difference between all these terms to a non-developer :)

**Comment by vladislav-kir** [comment]
The rule and a video are a nice summary of what each of them means :)

**Comment by AlexRothwell** [comment]
It was really great to see an explanation of the difference in these concepts, especially the data lake. I previously worked on an application for qualitative data analysis which looking back seems to have been used for analysing data puddles

**Comment by cvionnet** [comment]
Great video and rule, it helps to have a better understanding of the differences between data warehouses and data lakes 👍.
And it's good to see some love for the database side with tools like Copilot for Azure Data Studio.

**Comment by danielmackay** [comment]
I like the idea of learning more about data.  Personally, I haven't had a lot of exposure around data warehouses and data lakes.  Additionally, I would also like to learn more about reporting / analytics capabilities that sit on top of these.

**Comment by PennyWalker** [comment]
I really enjoyed the video, it was a great overview of the different options and I feel pretty comfortable talking about them now which is cool. I didn't really know the difference between a warehouse and a lake before so now I feel better prepared. :)

**Comment by jernejk** [comment]
I like to learn data from a common sense perspective, like "When you categorize data, exact dates usually aren't helpful, but the day of the week or the month of the year is!".

But nowadays, I dump data into Noteable, connect Noeable to ChatGPT-4 Plugins, say "Let there be sense of this data", and there be sense.

**Comment by chrisschultzssw** [comment]
The video was a great explanation of the differences between databases/warehouses/lakes.
Also love the inclusion of Azure certs :).

**Comment by Ben0189** [comment]
That is a great video to explain the terms of databases, warehouse, lakes. Given clear explanation to someone new to data. I'm interested to learn more about data analyze😊

**Comment by tombui99** [comment]
I did quite a bit of data warehouse projects back in uni and had a lot of fun using tools such as Tableau so this is a field I definitely have a lot of interest in! I will give the Azure cert a go in the near future 😁

**Comment by SamBWagner** [comment]
I did some subjects in Uni for big data and data in general. I've never engaged with any of it in the wild though. This rule (mainly the video) was a great introduction to what they are. I'd love to see some real-world implementations of the technology so I could understand it better.

Not an area I'm personally interested in but professionally I think it'd be great to learn about.

**Comment by JackDevAU** [comment]
I learnt about the general definitions of a Database, warehouse and lake. I think the video did a pretty good job at explaining everything! The rule however, has a lot of text maybe too much. I feel we should try finding a better way to display all this information, be that a different style of formatting rules or limiting text? I feel a good solution would also make it more consistent and easier to read across all rules.

**Comment by Harry-Ross** [comment]
Thought this was a great introduction to data warehouses and lakes - keen to learn more about them! I think the rule was a great summary of what we learned in the video - but would be better as a dot point summary.

**Comment by Levijj22** [comment]
Great video! Simple explanation.

**Comment by apforsyth** [comment]
Very enlightening especially re unstructured data in data lakes for AI

**Comment by lukecookssw** [comment]
Building and interacting with databases (and to an extent, data warehouses) is easy enough to visualize in my head, but I'm completely blind as to *how* you interact with something like a data lake where you are dealing with structured, semi-structured, and unstructured data all in the one location.

**Comment by KahaMason** [comment]
Data lakes seems interesting to me. I've used data warehouses before but I haven't used data lakes myself. Interested to learn more about them. 😄

**Comment by jackreimers** [comment]
I haven't used data warehouses or data lakes before but I'd be really eager to learn more, there seems to be a lot of interesting applications for them.

**Comment by amankumarrr** [comment]
Very well explained - Data warehouse and Data lake could be useful when we are processing big data for analytical purposes.

**Comment by AntPolkanov** [comment]
I worked quite a bit with SQL Server and NoSQL databases (DynamoDB, Realm), also have done some work on data normalization.

Does anyone normalize data higher than BCNF nowadays?

**Comment by steven0x51** [comment]
Great video and rule, very well explained the difference between database, data warehouse and data lake.

**Comment by wicksipedia** [comment]
![image](https://github.com/SSWConsulting/SSW.Rules.Content/assets/600044/e52ad315-4c65-4726-8900-5b76af5e9826)

**Comment by tomek-i** [comment]
I did not know the terminology before but encountered DB Warehouses before and know of a company who urgently need to implement that. 

If your report takes 1hr to generate and blocks every single user on your CRM solution then your alarm bells should go off.

I wonder how you work with Data Lakes. IRL, people dump anything into a lake to make it disappear and I wonder if that's the same for Data Lakes ? 🤔

**Comment by zacharykeeping** [comment]
Good video and rule - described the differences in data storage in an easy and concise way.

**Comment by matt-goldman** [comment]
My knowledge to data lakes is similar to Lake Torrens - dry and empty

**Comment by jimmy1992abc** [comment]
I am familiar with databases and data warehouses (not on the developer side), but the data lake is something new to me. Data lake basically means storing all data in one source. I got an idea for the future, build up a standard then give a random id to every document in the cloud, so no more duplicated cloud documents in the future and better verifiability and a saving of resources.

**Comment by AttackOnMorty** [comment]
My answer to the questions in the form is either None or No. Good to know about data storage.

**Comment by drwharris** [comment]
Ill have one of those, one of those, 2 of those and a few of those please. Now can you make them all work together!

My experience with data warehousing is people wanting to throw all their data into a common area with no thought and expect quality reports out of it. Hated it!

**Comment by duanxinhuan** [comment]
I love all of those!

**Comment by christoment** [comment]
I learned about Data base/warehouse/lake concept and differences 😄

**Comment by Dhruv-0987** [comment]
I want to learn more about non structured DB I think they are easier to use from a dev perspective.

**Comment by jerryluossw** [comment]
I love this rule, I think it's better to add a few specific usage scenarios for Data Lakes 😁

**Comment by jimmidier** [comment]
I'd love to learn more about NoSQL and Fabric. I tried to learn ELK stack once but stopped halfway. I wish I'd pick it up some time.

**Comment by AnkyChe** [comment]
Love the video, friendly for non-developers.

**Comment by sylhuang** [comment]
I have used both relational and document databases before, but I have never tried out data warehouses and data lakes. Both the video and the rule well explained the differences.

**Comment by Aibono1225** [comment]
Good video + rule, they are both very comprehensible. Good to know their usages and differences.

**Comment by tino-liu** [comment]
Need to learn more about structured databases before moving to non-structured one. Data Lake and its usage is cool but it feels like finding a needle in a haystack? Funny if a Data Ocean appears outa nowhere someday.

**Comment by johnxu16** [comment]
Like the jargon and comprehensible explanation of Data Lake and Data Warehouse.

Just to be curious, are there Data Ponds where I can cast my fishing rod 🤔

**Comment by JaneYangssw** [comment]
Great video and rules that helped me learn more about data lake and data warehouse. I didn't use these tools before😊.

**Comment by KristenHu** [comment]
Nice video, I never used data warehouse and data lake before, so I just keep tick no or none in the form🤣, but glad to know it today.

**Comment by william-liebenberg** [comment]
The CTF didn't consider the most important aspect of Data Warehouses when it comes to query performance... which is how the curated data is stored and what the common formats are. Row based querying is slow vs Columnar queries for big data is fast and furious! With Parquet as the open standard file format - which is a big party of why Microsoft Fabric is able to integrate with pretty much everything (vs something like Snowflake.

**Reply by lukecookssw** [reply]
We will be doing a separate CTF on data warehouses specifically, so we'll dive into more detail then.

**Comment by JeanThirion** [comment]
I wish I knew more about data... I feel ashamed I don't 😅

**Comment by ozair-ashfaque** [comment]
Interesting to know about differences and keen to learn them

**Comment by BrookJeynes** [comment]
Very accessible video and the rule explains things well

**Comment by TobyChurches** [comment]
Great overview video, definitely good for a basic understanding 🙂 It will be good to learn more about Data lakes and Data Warehouses

**Comment by sethdaily** [comment]
I previously did not know the difference between databases, warehouses, and lakes. Now I know! And they make much more sense.

**Comment by bahjatalaadelssw** [comment]
It's useful to have a overview of how data management is changing as the demand for information and content continues to grow at an exponential rate. I'll just live on my local NAS 🤣

**Comment by ozdemir-mehmet** [comment]
Need to replace DataWarehouse with DataOcean
Then we could have:
DataOcean (Top dog)
DataSea
DataLake
DataGlacier (Archive tier)
DataRiver 
DataStream

**Comment by Marxoz** [comment]
Show me the data!! 

Good topic this week. I think everyone should have used a database to understand how, at the least, unstructured data can be set up.
